text/0813-template-layering.md

```diff
@@ -0,0 +1,1001 @@
+and or deviate from the normal lexical scope capturing rules. It would be
+unfortunate if being restricted to the standard JavaScript environment also
+means dropping down to a completely different programming model for components.
+
+Currently, the addon [ember-template-imports][ember-template-imports] provides
+an alternative that does work in regular `.js`/`.ts` files – the "hbs backtick"
+format. However, it is not a good candidate for solving these problems for a
+few reasons, some of which were the same reasons RFC #779 chose not to go that
+route:
+
+[ember-template-imports]: https://github.com/ember-template-imports/ember-template-imports
+
+1. The feature was never specified in RFC #779 or any other RFC. It was added
+   to the addon when its purpose was to be a sandbox for experimentation. With
+   the acceptance of RFC #779, the experimentation phase is over and the addon
+   should be transitioned into a compliant polyfill, which means removing the
+   alternative `hbs` form.
+
+2. The naming conflicts with the "official" `hbs` from `ember-cli-htmlbars`.
+   This would have been fine if we chose to go that route in RFC #779 and have
+   plans to evolve the official version to have the same feature. With RFC #779
+   favoring the `<template>` approach, this is not going to happen and this
+   conflict will now be quite confusing, especially since they actually do very
+   different things (e.g. the "official" `hbs` does not return a component).
+
+3. It uses a standard JavaScript syntax in a non-standard way, semantics-wise.
+   It looks like an inert JavaScript string but has access to the lexical scope
+   around it (without using the `${ ... }` syntax), which can be confusing. It
+   also means that it is impossible to provide a runtime implementation using
+   the same syntax that works in environments that does not permit a build step.
+
+4. It uses `static template = ` to associate templates to classes. This implies
+   runtime behavior that is actually not true (e.g. TypeScript will believe the
+   `template` property exists on the class).
+
+5. Values consumed by the template but is otherwise unused in the rest of the
+   file may generate errors/warnings from linters or language servers.
+
+To address these issues, this RFC propose that we introduce a new `template()`
+function that servers as a middle ground between the `<template>` language
+extension and the low-level primitives.
+
+- It is designed to have the same semantics as the `<template>` feature (such
+  as the strict-mode opt-in, returning a template-only component when not
+  attached to a class), providing the same high-level programming model for
+  authoring components in standard JavaScript environments.
+
+- It requires manually supplying the lexical scope variable bindings (the
+  "scope function").
+
+- It uses a [static initializer block][static-block] to associate the template.
```

Comment:
Based on reading the Class code snippets, the proposal is to use the static initializer block to side effect the association of the template?  Or is there actual static initialization on the class itself?  Perhaps inconsequential, but was hoping to enhance my understanding of this proposal.